snuba/settings/__init__.py

```diff
@@ -14,12 +14,12 @@
 ENABLE_DEV_FEATURES = os.environ.get("ENABLE_DEV_FEATURES", False)
 
 DEFAULT_DATASET_NAME = "events"
-DISABLED_DATASETS: Set[str] = set()
+DISABLED_DATASETS: Set[str] = {"metrics"}
 
 # Clickhouse Options
 CLICKHOUSE_MAX_POOL_SIZE = 25
 
-DISABLED_STORAGE_SETS: Set[str] = set()
+DISABLED_STORAGE_SETS: Set[str] = {"metrics"}
```

Comment:
Ideally, I think these wouldn't be independently configurable for those datasets and storage sets that apply specifically to the dev feature. Otherwise the user could, say, remove metrics from here and also set ENABLE_DEV_FEATURES to false, which may lead to weird things happening.

What do you think about hardcoding the dev specific datasets and storage sets and automatically enabling and disabling them based solely on the `ENABLE_DEV_FEATURES` flag?

Reply:
I think this is a good point. 
I will change the PR below this so that a DEV dataset is automatically disabled unless ENABLE_DEV_FEATURES is on. 
I don't think we can remove DISABLED_DATASETS as we may disable datasets for reasons other than dev. See the `querylog` dataset.

About the storage set, do you mean having the list of dev storage sets hardcoded like for datasets, entity and storage instead of it being configurable ? That would make sense to me.

Reply:
done